cores/esp32/esp32-hal-rgb-led.c

```diff
@@ -0,0 +1,51 @@
+#include "esp32-hal-rgb-led.h"
+
+#ifdef BOARD_HAS_NEOPIXEL
+
+void RGBLedWrite(uint8_t pin, uint8_t red_val, uint8_t green_val, uint8_t blue_val){
+  log_d("RGB: %d %d %d", red_val, green_val, blue_val);
+  rmt_data_t led_data[24];
+  static rmt_obj_t* rmt_send = NULL;
+  static bool initialized = false;
+
+  uint8_t _pin;
+  if(pin == LED_BUILTIN){
```

Comment:
this should be evaluated only if the board has neopixel, else LED_BUILTIN would be lower than SOC_GPIO_PIN_COUNT

Reply:
The entire function was safeguarded `#ifdef BOARD_HAS_NEOPIXEL`
But I will remove safeguard for the entire function a wrap only this.